tests/example_messenger.go

```diff
@@ -0,0 +1,97 @@
+package tests
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/accounts/abi/bind"
+	"github.com/ava-labs/subnet-evm/core/types"
+	teleportermessenger "github.com/ava-labs/teleporter/abi-bindings/go/Teleporter/TeleporterMessenger"
+	"github.com/ava-labs/teleporter/tests/network"
+	"github.com/ava-labs/teleporter/tests/utils"
+	localUtils "github.com/ava-labs/teleporter/tests/utils/local-network-utils"
+	. "github.com/onsi/gomega"
+)
+
+func ExampleMessengerGinkgo() {
+	ExampleMessenger(&network.LocalNetwork{})
+}
+
+func ExampleMessenger(network network.Network) {
+	var (
+		teleporterMessageID *big.Int
+	)
+
+	subnets := network.GetSubnetsInfo()
+	subnetAInfo := subnets[0]
+	subnetBInfo := subnets[1]
+	teleporterContractAddress := network.GetTeleporterContractAddress()
+	fundedAddress, fundedKey := network.GetFundedAccountInfo()
+
+	subnetATeleporterMessenger, err := teleportermessenger.NewTeleporterMessenger(
+		teleporterContractAddress, subnetAInfo.ChainRPCClient,
+	)
+	Expect(err).Should(BeNil())
+	subnetBTeleporterMessenger, err := teleportermessenger.NewTeleporterMessenger(
+		teleporterContractAddress, subnetBInfo.ChainRPCClient,
+	)
+	Expect(err).Should(BeNil())
+
+	//
+	// Deploy ExampleMessenger to Subnets A and B
+	//
+	ctx := context.Background()
+
+	_, subnetAExampleMessenger := localUtils.DeployExampleCrossChainMessenger(ctx, fundedAddress, fundedKey, subnetAInfo)
+	exampleMessengerContractB, subnetBExampleMessenger := localUtils.DeployExampleCrossChainMessenger(
+		ctx, fundedAddress, fundedKey, subnetBInfo,
+	)
+
+	//
+	// Call the example messenger contract on Subnet A
+	//
+	message := "Hello, world!"
+	optsA := utils.CreateTransactorOpts(ctx, subnetAInfo, fundedAddress, fundedKey)
+	tx, err := subnetAExampleMessenger.SendMessage(
+		optsA,
+		subnetBInfo.BlockchainID,
+		exampleMessengerContractB,
+		fundedAddress,
```

Comment:
think it's confusing to use fundedAddress for fee contract address here, think we can use `address(0)` since fee amount is 0